### DISCUSSION TITLE
Soldering Question

### DISCUSSION THREAD

**Discussion by jamesphw** [emporia-vue-local/esphome#256] [Q&A]
I've noticed that the via holes on the emporia vue are pre-filled with solder (in the spot where you'd normally put the male pins to attach for programming). So I have a basic question -- did you melt that solder out and put in fresh, or simply push your pins into the existing solder? Thanks for the help! I've never had to deal with the pre-filled holes, I assume you just wipe clean, flux, heat the pin, and insert?

**Comment by jamesphw** [comment]
I'm already answering my own question, but I guess I will just remove what's there already. Though would still like to know if I can skip this step and use what's in the hole already.

**Reply by flaviut** [reply]
It's simply a matter of physical possibility. It's impossible to melt 6 pins simultaneously, so you can either remove the solder and put in the whole row, or break off each pin and solder them individually.  first option is far easier and less fiddly IMO

**Reply by supermarkert** [reply]
I had individual pins, and just heated and shoved through each pin. Seemed easier than trying to remove the solder.

**Reply by jamesphw** [reply]
@supermarkert Thanks for the advice. I will just go for that instead then! I just needed confirmation that the material that's in the holes already is workable for the connection.
